src/device/usbd.c

```diff
@@ -61,6 +61,9 @@ typedef struct {
 
 static usbd_device_t _usbd_dev = { 0 };
 
+// Invalid driver ID in itf2drv[] ep2drv[][] mapping
+enum { DRVID_INVALID = 0xff };

```

Comment:
I'm not sure if enum is good or bad; I guess it's a matter of opinion.
In C, we can't declare the enum of any particular type; I think it ends up being an integer or unsigned integer? Would just declaring it as a "uint8_t const" work better since it would be typed? (I don't have a strong opinion on this)

Reply:
uint8_t const is still addressable and will take up 1 byte of flash. Not much but It isn't worthy it. Did your compiler complain, we can change it to macro instead

Reply:
Since it's const, the compiler knows it won't change. It won't do any load/store for it; the immediate value is directly encoded in the opcode. It's treated the same exact way as the #define by the compiler; just it would have type checking.

Reply:
It is still located in addressable flash and occupy storage. Enum and macro define is translated to immediate constant and won't cost anything. I will change the enum to unsigned as you suggested.